code/modules/power/prototype_shield_emitter.dm

```diff
@@ -0,0 +1,153 @@
+		return TRUE
+	if(!anchored)
+		anchored = TRUE
+		to_chat(user, "<span class='warning'>You bolt the [src] to the floor!</span>")
+	else
+		anchored = FALSE
+		to_chat(user, "<span class='warning'>You unbolt the [src] from the floor!</span>")
+	return TRUE
+
+/obj/machinery/power/proto_sh_emitter/process()
+	var/area/a = get_area(src)
+	var/turf/Turf = get_turf(src)
+	if(a.power_equip == FALSE)
+		is_on = FALSE
+		update_icon_state()
+		message_admins("Prototype Shield Emitter turned off at [ADMIN_VERBOSEJMP(Turf)]")
+		log_game("Prototype Shield Emitter turned off at [AREACOORD(Turf)]")
+
+/obj/machinery/power/proto_sh_emitter/interact(mob/user)
+	var/turf/T = /turf
+	var/turf/Turf = get_turf(src)
+	var/list/outline = list()
+	var/list/internal = list()
+	var/area/a = get_area(src)
+	add_fingerprint(user)
+	if(!anchored)
+		to_chat(user, "<span class='warning'>You need to anchor the [src] first!</span>")
+		return
+	if(a.power_equip == FALSE)
+		to_chat(user, "<span class='warning'>There is no power in this area!!</span>")
+		return
+	if(locked)
+		to_chat(user, "<span class='warning'>The controls are locked!</span>")
+		return
+	if(is_on == TRUE)
+		to_chat(user, "<span class='warning'>You turn off the [src] and the generated shields!</span>")
+		message_admins("Prototype Shield Emitter turned off at [ADMIN_VERBOSEJMP(Turf)]")
+		log_game("Prototype Shield Emitter turned off at [AREACOORD(Turf)]")
+		is_on = FALSE
+		for(var/H in signs)
+			qdel(H)
+		update_icon_state()
+	else
+		to_chat(user, "<span class='warning'>You turn on the [src] and the generated shields!</span>")
+		message_admins("Prototype Shield Emitter turned on at [ADMIN_VERBOSEJMP(Turf)]")
+		log_game("Prototype Shield Emitter turned on at [AREACOORD(Turf)]")
+		is_on = TRUE
+		update_icon_state()
+		switch(dir) //this part check the direction of the machine and create the block in front of it
+			if(NORTH)
+				for(T in block(locate(src.x - 2, src.y + 1, src.z), locate(src.x + 2, src.y + 5, src.z)))
+					outline += T
+				for(T in block(locate(src.x - 1, src.y + 2, src.z), locate(src.x + 1, src.y + 4, src.z)))
+					outline -= T
+				for(T in outline)
+					new /obj/machinery/holosign/barrier/power_shield/wall(T, src)
+				for(T in block(locate(src.x - 1, src.y + 2, src.z), locate(src.x + 1, src.y + 4, src.z)))
+					internal += T
+				for(T in internal)
+					new /obj/machinery/holosign/barrier/power_shield/floor(T, src)
+			if(SOUTH)
+				for(T in block(locate(src.x - 2, src.y - 1, src.z), locate(src.x + 2, src.y - 5, src.z)))
+					outline += T
+				for(T in block(locate(src.x - 1, src.y - 2, src.z), locate(src.x + 1, src.y - 4, src.z)))
+					outline -= T
+				for(T in outline)
+					new /obj/machinery/holosign/barrier/power_shield/wall(T, src)
+				for(T in block(locate(src.x - 1, src.y - 2, src.z), locate(src.x + 1, src.y - 4, src.z)))
+					internal += T
+				for(T in internal)
+					new /obj/machinery/holosign/barrier/power_shield/floor(T, src)
```

Comment:
It looks like what you're subtracting from outline and the contents of internal are about the same, could you do both of these things at once?

Reply:
yeah i actually can, i just add `internal +T` to the second for(block()) loop so that i can remove the third for(block()) loop